labs/sysdev-buildroot-stm32mp2/sysdev-buildroot-stm32mp2.tex

```diff
@@ -0,0 +1,577 @@
+cp ../../kernel/linux/0001-Custom-DTS-for-Bootlin-lab.patch \
+   board/bootlin/training/
+\end{bashinput}
+
+We now need to download additional files so that Buildroot can support the board.
+To do this, we will fetch the sources provided by Bootlin.
+
+\begin{bashinput}
+$ cd ..
+$ git clone https://github.com/bootlin/buildroot-external-st.git
+\end{bashinput}
+
+We will configure Buildroot to use this kernel configuration.
+
+\section{Configure Buildroot}
+
+In our case, we would like to:
+
+\begin{itemize}
+\item Generate an embedded Linux system for ARM;
+\item Use an already existing external toolchain instead of having
+  Buildroot generating one for us;
+\item Compile the Linux kernel and deploy its modules in the root
+  filesystem;
+\item Integrate {\em BusyBox}, {\em alsa-utils},
+  {\em mpd}, {\em mpc} and {\em evtest} in our embedded Linux system;
+\item Integrate the target filesystem into a tarball
+\end{itemize}
+
+To run the configuration utility of Buildroot, simply run:
+
+\bashcmd{$ make menuconfig}
+
+Set the following options. Don't hesitate to press the \code{Help}
+button whenever you need more details about a given option:
+
+\begin{itemize}
+\item \code{Target options}
+  \begin{itemize}
+  \item \code{Target Architecture}: \code{AArch64 (little endian)}
+  \item \code{Target Architecture Variant}: \code{cortex-A53}
+  \item \code{Target ABI}: \code{EABIhf}
+  \item \code{Floating point strategy}: \code{FP-ARMv8}
+  \end{itemize}
+\item \code{Toolchain}
+  \begin{itemize}
+  \item \code{Toolchain type}: \code{External toolchain}
+  \item \code{Toolchain}: \code{Custom toolchain}
+  \item \code{Toolchain path}: use the toolchain you built:
+  \code{/home/<user>/x-tools/aarch64-training-linux-musl}
+    (replace \code{<user>} by your actual user name)
+  \item \code{External toolchain gcc version}: \code{13.x}
+  \item \code{External toolchain kernel headers series}: \code{6.13}
+  \item \code{External toolchain C library}: \code{musl (experimental)}
+  \item We must tell Buildroot about our toolchain configuration, so
+    select \code{Toolchain has SSP support?} and
+    \code{Toolchain has C++ support?}.
+    Buildroot will check these parameters anyway.
+  \end{itemize}
+
+\item \code{Build options}
+  \begin{itemize}
+  \item Set \code{global patch and hash direct} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/common/patches}
+  \item Set \code{Root filesystem overlay directories} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/stm32mp2/overlay}
+  \item Set \code{Custom scripts to run before creating filesystem images} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/common/post-build.sh}
+  \item Set \code{Custom scripts to run after creating filesystem images} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/common/post-image.sh}
+  \item Set \code{Extra arguments passed to custom scripts} to \code{$(BR2_EXTERNAL_ST_PATH)/board/stmicroelectronics/stm32mp2/genimage.cfg}
```

Comment:
No, we really don't want to use buildroot-external-st. We don't need to build a full SD card image. Check how the lab is done for stm32mp1.